src/datachain/data_storage/warehouse.py

```diff
@@ -235,7 +229,7 @@ def dataset_select_paginated(
                 processed = False
                 for row in results:
                     processed = True
-                    yield RowDict(zip(cols_names, row))
```

Comment:
is it the main change here? (avoiding creating dicts?)

Reply:
Yes, it is one of the major changes in this PR. There are three major things:

- return rows instead of dicts from SQL query (reduce memory consumption) **this one**
- marshal UDF rows with msgpack instead of dill (speedup pickling/types conversion)
- add batching to UDF results (speedup writing/reading to/from multiprocessing queues)

Reply:
can it be split into 3 PRs (chained or completely separate) + some cleanup? it would way easier to read, better history, less discussions on each :) 

(it's fine as it is now - I read almost all of it by now, but it took some time, and I feel we could have merged almost immediately some things, while others deserve some discussion)

Reply:
Let me think 👍